lib/ClangImporter/SwiftLookupTable.h

```diff
@@ -45,6 +45,102 @@ class TypedefNameDecl;
 
 namespace swift {
 
+/// A name from a SwiftLookupTable that has not been deserialized into a
+/// DeclBaseName yet
+struct SerializedSwiftName {
+  /// The kind of the name if it is a special name
+  DeclBaseName::Kind Kind;
+  /// The name of the idenifier if it is not a special name
+  StringRef Name;
+
+  SerializedSwiftName() : Kind(DeclBaseName::Kind::Normal), Name(StringRef()) {}
+
+  explicit SerializedSwiftName(DeclBaseName::Kind Kind)
+      : Kind(Kind), Name(StringRef()) {}
+
+  explicit SerializedSwiftName(StringRef Name)
+      : Kind(DeclBaseName::Kind::Normal), Name(Name) {}
+
+  SerializedSwiftName(DeclBaseName BaseName) {
+    Kind = BaseName.getKind();
+    if (BaseName.getKind() == DeclBaseName::Kind::Normal) {
+      Name = BaseName.getIdentifier().str();
+    }
+  }
+
+  /// Deserialize the name, adding it to the context's identifier table
+  DeclBaseName toDeclBaseName(ASTContext &Context) const;
+
+  bool empty() const {
+    return Kind == DeclBaseName::Kind::Normal && Name.empty();
+  }
+
+  /// Return a string representation of the name that can be used for sorting
+  StringRef sortStr() const {
```

Comment:
Nitpick: I think this should be `private` for now. This partially comes from me still not being a fan of using the "str" abbreviation with any kind of description.

(also, missing period again)